pypowsybl/network.py

```diff
@@ -5327,5 +5342,53 @@ def replace_tee_point_by_voltage_level_on_line(network: Network, tee_point_line1
     if new_line2_name is None:
         new_line2_name = new_line2_id
 
-    _pp.replace_tee_point_by_voltage_level_on_line(network._handle, tee_point_line1, tee_point_line2, tee_point_line_to_remove,
-        bbs_or_bus_id, new_line1_id, new_line1_name, new_line2_id, new_line2_name)
+    _pp.replace_tee_point_by_voltage_level_on_line(network._handle, tee_point_line1, tee_point_line2,
+                                                   tee_point_line_to_remove,
+                                                   bbs_or_bus_id, new_line1_id, new_line1_name, new_line2_id,
+                                                   new_line2_name)
+
+
+def create_voltage_level_topology(network: Network, df: _DataFrame = None, raise_exception: bool = False,
+                                  reporter: _Reporter = None, **kwargs: _ArrayLike) -> None:
+    """
+    Creates the topology of a given symmetrical voltage level, containing a given number of busbar with a given number of sections.
+
+    Args:
+        network: the network in which the busbar sections are.
+        df: Attributes as a dataframe.
+        raise_exception: whether an exception should be raised if a problem occurs. By default, false.
+        reporter: an optional reporter to get functional logs.
+        kwargs: attributes as keyword arguments.
+    Notes:
+        The voltage level must be created and in node/breaker topology.
+        Busbar sections will be created, as well as disconnectors or breakers between each section depending on the
+        switch_kind list.
+
+        The input dataframe expects these attributes:
+        - **voltage_level_id**: the identifier of the voltage level where the topology should be created.
+        - **low_busbar_index**: the lowest busbar index to be used. By default, 1 (no other busbar sections).
+        - **busbar_count**: the total number of busbar to be created.
+        - **low_section_index**: the lowest section index to be used. By default, 1.
+        - **busbar_section_prefix_id**: an optional prefix to put on the names of the created busbar sections. By
+        default, nothing.
+        - **switch_prefix_id**: an optional prefix to put on the names of the created switches. By default, nothing.
+        - **switch_kinds**: string or list containing the type of switch between each section. It should contain
+        section_count - 1 switches and should look like that 'BREAKER, DISCONNECTOR' or ['BREAKER', 'DISCONNECTOR'].
+
+    Examples:
+
+    .. code-block:: python
+        pp.network.create_voltage_level_topology(network=network, raise_exception=True, id='VL', busbar_count=3,
+                                                 section_count=3, switch_kinds='BREAKER, DISCONNECTOR')
+    """
+    metadata = _pp.get_network_modification_metadata(NetworkModificationType.VOLTAGE_LEVEL_TOPOLOGY_CREATION)
+    df = _adapt_df_or_kwargs(metadata, df, **kwargs)
+    for i in range(len(df)):
```

Comment:
This kind of for loop can be VERY slow in python, we should try to avoid it as much as possible.
Unfortunately I am not sure you can apply a python method on each element in a really more efficient way.
It might be slightly better to use [Series.transform](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.transform.html) at least, but maybe not a big difference.

Since this network modification will never be a performance bottleneck, I think, let's keep it this way if there is no other solution.

Reply:
maybe with map method I can do something